app/library/class/page.tsx

```diff
@@ -0,0 +1,235 @@
+"use client"
+
+import React, { useState } from 'react';
+import { useRouter } from 'next/navigation';
+import BottomNavigationBar from '@/components/BottomNavigationBar';
+import Loading from '../../loading';
+import TopBar from '@/components/TopBar';
+import PrimaryButton from '@/components/Button';
+import { getSubjects, getClassChapters, getGrades, getResourcesOfChapter, getTeachers } from '../../../api/afdb/library';
+import { Chapter, Resource, Topic, Teacher } from '../../types';
+import { useEffect } from 'react';
+import Link from 'next/link';
+import ExpandIcon from "../../../assets/expand.png";
+import CollapseIcon from "../../../assets/collapse.png";
+import PlayIcon from "../../../assets/play.png";
+import BackIcon from "../../../assets/icon.png";
+import Image from 'next/image';
+import { useSearchParams } from 'next/navigation';
+
+const ClassLibrary = () => {
+    const [activeTab, setActiveTab] = useState('Physics');
+    const [chapters, setChapters] = useState<Chapter[]>([]);
+    const [resources, setResources] = useState<Resource[]>([]);
+    const [expandedChapters, setExpandedChapters] = useState<Record<number, boolean>>({});
+    const [selectedGrade, setSelectedGrade] = useState(11);
+    const [teachers, setTeachers] = useState<Teacher[]>([]);
+    const [selectedTeacher, setSelectedTeacher] = useState<number | undefined>(undefined);
+    const [selectedChapter, setSelectedChapter] = useState<number | null>(null);
+    const [chapterList, setChapterList] = useState<Chapter[]>([]);
+    const gradeOptions = [11, 12];
+    const router = useRouter();
+    const searchParams = useSearchParams();
+    const [isLoading, setIsLoading] = useState(true);
+    const selectedCourse = searchParams.get('course');
+    const neetSubjects = ['Physics', 'Chemistry', 'Biology'];
+    const jeeSubjects = ['Physics', 'Chemistry', 'Maths'];
+
+    const handleTabClick = async (tabName: string) => {
```

Comment:
Just a thought -- this is probably highly common code between class and content library, and can be moved into a common component with variables?

Reply:
yes yes